doc/connectivity/networking/net_config_guide.rst

```diff
@@ -0,0 +1,273 @@
+  Maximum number of TLS/DTLS ciphersuites per socket.
+  This variable sets maximum number of TLS/DTLS ciphersuites that can
+  be used with specific socket, if set explicitly by socket option.
+  By default, all ciphersuites that are available in the system are
+  available to the socket.
+
+:kconfig:option:`CONFIG_NET_SOCKETS_TLS_MAX_APP_PROTOCOLS`
+  Maximum number of supported application layer protocols.
+  This variable sets maximum number of supported application layer
+  protocols over TLS/DTLS that can be set explicitly by a socket option.
+  By default, no supported application layer protocol is set.
+
+:kconfig:option:`CONFIG_NET_SOCKETS_TLS_MAX_CLIENT_SESSION_COUNT`
+  This variable specifies maximum number of stored TLS/DTLS sessions,
+  used for TLS/DTLS session resumption.
+
+:kconfig:option:`CONFIG_TLS_MAX_CREDENTIALS_NUMBER`
+   Maximum number of TLS credentials that can be registered.
+   Make sure that this value is high enough so that all the
+   certificates can be loaded to the store.
+
+
+IPv4/6 Options
+**************
+
+:kconfig:option:`CONFIG_NET_IF_MAX_IPV4_COUNT`
+   Maximum number of IPv4 network interfaces in the system.
+   This tells how many network interfaces there will be in the system
+   that will have IPv4 enabled.
+   For example if you have two network interfaces, but only one of them
+   can use IPv4 addresses, then this value can be set to 1.
+   If both network interface could use IPv4, then the setting should be
+   set to 2.
+
+:kconfig:option:`CONFIG_NET_IF_MAX_IPV6_COUNT`
+   Maximum number of IPv6 network interfaces in the system.
+   This is similar setting as the IPv4 count option but for IPv6.
+
+
+TCP Options
+***********
+
+:kconfig:option:`CONFIG_NET_TCP_TIME_WAIT_DELAY`
+  How long to wait in TCP *TIME_WAIT* state (in milliseconds).
+  To avoid a (low-probability) issue when delayed packets from
+  previous connection get delivered to next connection reusing
+  the same local/remote ports,
+  `RFC 793 <https://www.rfc-editor.org/rfc/rfc793>`_ (TCP) suggests
+  to keep an old, closed connection in a special *TIME_WAIT* state for
+  the duration of 2*MSL (Maximum Segment Lifetime). The RFC
+  suggests to use MSL of 2 minutes, but notes
+
+  *This is an engineering choice, and may be changed if experience indicates
+  it is desirable to do so.*
+
+  For low-resource systems, having large MSL may lead to quick
+  resource exhaustion (and related DoS attacks). At the same time,
+  the issue of packet misdelivery is largely alleviated in the modern
+  TCP stacks by using random, non-repeating port numbers and initial
+  sequence numbers. Due to this, Zephyr uses much lower value of 1500ms
+  by default. Value of 0 disables *TIME_WAIT* state completely.
+
+:kconfig:option:`CONFIG_NET_TCP_RETRY_COUNT`
+  Maximum number of TCP segment retransmissions.
+  The following formula can be used to determine the time (in ms)
+  that a segment will be be buffered awaiting retransmission::
```

Comment:
You probably missed it in the previous comments (my bad for not having making it clearer) but the extra ":" needs to go otherwise following block is treated as a literal code block. Good to go otherwise, thanks!

```suggestion
  that a segment will be be buffered awaiting retransmission:
```